fvm/blueprints/scheduled_callback.go

```diff
@@ -0,0 +1,125 @@
+package blueprints
+
+import (
+	_ "embed"
+	"fmt"
+	"strings"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/encoding/ccf"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// processScheduledCallbacksTransaction calls scheduled callback contract
+// and process new callbacks that should be executed.
+//
+//go:embed scripts/processScheduledCallbacksTransaction.cdc
+var processCallbacksTransaction string
+
+// executeCallbacksTransaction calls scheduled callback contract
+// to execute the provided callback by ID.
+//
+//go:embed scripts/executeScheduledCallbackTransaction.cdc
+var executeCallbacksTransaction string
+
+const (
+	placeholderScheduledContract     = "import \"CallbackScheduler\""
+	processedCallbackIDFieldName     = "ID"
+	processedCallbackEffortFieldName = "executionEffort"
+	processedEventTypeTemplate       = "A.%v.CallbackScheduler.CallbackProcessed"
+)
+
+func ProcessCallbacksTransaction(chain flow.Chain) *flow.TransactionBody {
+	script := prepareScheduledContractTransaction(chain, processCallbacksTransaction)
+
+	return flow.NewTransactionBody().
+		SetScript(script).
+		SetComputeLimit(SystemChunkTransactionGasLimit)
+}
+
+func ExecuteCallbacksTransactions(chainID flow.Chain, processEvents flow.EventsList) ([]*flow.TransactionBody, error) {
+	txs := make([]*flow.TransactionBody, 0, len(processEvents))
+
+	for _, event := range processEvents {
+		id, effort, err := callbackArgsFromEvent(event)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get callback args from event: %w", err)
+		}
+
+		tx := executeCallbackTransaction(chainID, id, effort)
+		txs = append(txs, tx)
+	}
+
+	return txs, nil
+}
+
+func executeCallbackTransaction(chain flow.Chain, id []byte, effort uint64) *flow.TransactionBody {
+	script := prepareScheduledContractTransaction(chain, executeCallbacksTransaction)
+	return flow.NewTransactionBody().
+		SetScript(script).
+		AddArgument(id).
+		SetComputeLimit(effort)
+}
+
+// callbackArgsFromEvent decodes the event payload and returns the callback ID and effort.
+//
+// The event for processed callback event is emitted by the process callback transaction from
+// callback scheduler contract and has the following signature:
+// event CallbackProcessed(ID: UInt64, executionEffort: UInt64)
+func callbackArgsFromEvent(event flow.Event) ([]byte, uint64, error) {
+	scheduledContractAddress := "0x0000000000000000" // todo use contract addr
```

Comment:
we need to clear this up before we can merge this.

Reply:
the problem I have is that before this is part of the core-contracts and bootstrap I can't define this, what do you suggest? maybe I'm missing something